src/controller/oauth.js

```diff
@@ -267,37 +268,28 @@ const token = async (req, res) => {
     }
 
     if (codeVal.session.code_challenge !== cv) {
-      return res.status(401).send('Invalid code_verifier');
+      throw new JacksonError('Invalid code_verifier');
     }
   } else if (codeVal && codeVal.session) {
-    return res
-      .status(401)
-      .send('Please specify client_secret or code_verifier');
+    throw new JacksonError('Please specify client_secret or code_verifier');
   }
 
   // store details against a token
   const token = crypto.randomBytes(20).toString('hex');
 
   await tokenStore.put(token, codeVal.profile);
 
-  res.json({
+  return {
     access_token: token,
     token_type: 'bearer',
     expires_in: options.db.ttl,
-  });
+  };
 };
 
-const userInfo = async (req, res) => {
-  let token = extractAuthToken(req);
```

Comment:
The token extraction needs to move into jackson.js

Reply:
@devkiran Apologies, the above statement was misleading now that I read it again. I meant the token extraction itself as opposed to the function, thanks for sorting that out